module/observations/helper.js

```diff
@@ -1048,4 +1054,176 @@ module.exports = class ObservationsHelper {
+        return new Promise(async (resolve, reject) => {
+
+            try {
+
+                if (link == "") {
+                    throw new Error(messageConstants.apiResponses.LINK_REQUIRED_CHECK)
+                }
+
+                let observationSolutionData = await solutionHelper.solutionDocuments({
+                    link: link,
+                    type : messageConstants.common.OBSERVATION,
+                    isReusable: false,
+                    status: messageConstants.common.ACTIVE_STATUS,
+                    endDate : {$gte: new Date()}
+                });
+
+                if(!Array.isArray(observationSolutionData) || observationSolutionData.length < 1){
+                    return resolve({
+                        message: messageConstants.apiResponses.LINK_IS_EXPIRED,
+                        result: []
+                    });
+                   
+                }
+
+                let observationData = await this.observationDocuments({
+                    solutionExternalId : observationSolutionData[0].externalId
+                });
+
+                if(observationData){
+                    return resolve({
+                        message: messageConstants.apiResponses.OBSERVATION_LINK_VERIFIED,
+                        result: observationData
+                    });
+                }
+
+                
+                let entities = [];
+                if(userId){
+                    let userEntities = await userExtensionHelper.getUserEntities(userId);
+                    if(userEntities.length > 0){
+                        let entityIdsWithSolutionSubType = await entitiesHelper.entityDocuments({
+                            _id :  { $in : userEntities},
+                            entityType : observationSolutionData[0].subType
+                        }, [
+                            "_id"
+                        ]);
+
+                        for(let pointerToUserExtension = 0; pointerToUserExtension < entityIdsWithSolutionSubType.length; 
+                            pointerToUserExtension++){
+                            entities.push(entityIdsWithSolutionSubType[pointerToUserExtension]._id)
+                        }
+                    }
+                }
+                
+
+                let solutionId = observationSolutionData[0]._id;
+                let programId = observationSolutionData[0].programId;
+                let today = new Date();
+                let startDate= moment(today).format("YYYY-MM-DD");
+                let endDate = moment(startDate, "YYYY-MM-DD").add('years', 1).format("YYYY-MM-DD");
+                let dataObj = {
+                    "name": observationSolutionData[0].name ,
+                    "description": observationSolutionData[0].description,
+                    "startDate": startDate,
+                    "endDate": endDate,
+                    "status": messageConstants.common.ACTIVE_STATUS,
+                    "entities": entities
```

Comment:
@priyanka-TL - Store the link value in observation to track which observation are created from link, also add the key in model and make it an index